src/pb_stub.cc

```diff
@@ -539,43 +540,48 @@ Stub::ProcessResponse(InferResponse* response)
 void
 Stub::LoadGPUBuffers(std::unique_ptr<IPCMessage>& ipc_message)
 {
-  AllocatedSharedMemory<char> gpu_buffers_handle =
-      shm_pool_->Load<char>(ipc_message->Args());
+  ScopedDefer load_gpu_buffer_response([this] {
+    // LoadGPUBuffers must let the parent process know when loading the
+    // buffers have been finished.
+    parent_message_queue_->Push(DUMMY_MESSAGE);
+    gpu_tensors_.clear();
+  });
 
-  uint64_t* gpu_buffer_count =
-      reinterpret_cast<uint64_t*>(gpu_buffers_handle.data_.get());
-  bi::managed_external_buffer::handle_t* gpu_buffers_handle_shm =
-      reinterpret_cast<bi::managed_external_buffer::handle_t*>(
-          gpu_buffers_handle.data_.get() + sizeof(uint64_t));
-
-  if (gpu_tensors_.size() != *gpu_buffer_count) {
-    LOG_INFO
-        << (std::string(
-                "GPU buffers size does not match the provided buffers: ") +
-            std::to_string(gpu_tensors_.size()) +
-            " != " + std::to_string(*gpu_buffer_count));
-    return;
+  AllocatedSharedMemory<GPUBuffersShm> gpu_buffers_handle =
```

Comment:
could we use GPUBuffers as the structure name (leave off the Shm suffix)?

Reply:
Responded here: https://github.com/triton-inference-server/python_backend/pull/249#discussion_r1214774976